example/echo-op/echo_op.cpp

```diff
@@ -240,10 +222,10 @@ async_echo(
                         // using available capacity in the buffer first.
 
                         bytes_to_read = std::min<std::size_t>(
-                              std::max<std::size_t>(512,                // under 512 is too little,
-                                  buffer_.capacity() - buffer_.size()),
-                              std::min<std::size_t>(65536,              // and over 65536 is too much.
-                                  buffer_.max_size() - buffer_.size()));
+                            std::max<std::size_t>(512,                // under 512 is too little,
+                                                  buffer_.capacity() - buffer_.size()),
+                            std::min<std::size_t>(65536,              // and over 65536 is too much.
+                                                  buffer_.max_size() - buffer_.size()));
```

Comment:
\sigh

Reply:
An unfortunate oversight due to ide autoformatting whil copy/pasting the code. In order to compile it could no longer be a template class.

Reply:
An unfortunate oversight due to ide autoformatting whil copy/pasting the code. In order to compile it could no longer be a template class.